listenbrainz/webserver/static/js/src/AlertNotificationsHOC.test.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from "react";
+import { shallow, mount } from "enzyme";
+import { withAlertNotifications } from "./AlertNotificationsHOC";
+
+const fakeComponent = function FakeComponent() {
+  return <div />;
+};
+
+describe("AlertNotifications higher-order component", () => {
+  beforeAll(() => {
+    // For alert id
+    jest.spyOn(Date.prototype, "getTime").mockImplementation(() => 0);
+  });
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("renders alerts on screen with the correct css classname", () => {
+    const WrappedComponent = withAlertNotifications(fakeComponent);
+    const fakeAlerts = [
+      {
+        id: 0,
+        type: "warning" as AlertType,
+        headline: "Test",
+        message: "Have you seen the Fnords?",
+        count: 1,
+      },
+      {
+        id: 1,
+        type: "danger" as AlertType,
+        headline: "Oh no!",
+        message: "The foo hit the bar",
+        count: 1,
+      },
+      {
+        id: 2,
+        type: "success" as AlertType,
+        headline: "Oh yeah!",
+        message: "Time for *us* to hit the bar",
+        count: 1,
+      },
+    ];
+    const wrapper = mount<React.ComponentType>(
+      <WrappedComponent initialAlerts={fakeAlerts} />
+    );
+
+    expect(wrapper.find(".alert")).toHaveLength(3);
+    expect(wrapper.find(".alert").at(0).hasClass("alert-warning")).toBeTruthy();
+    expect(wrapper.find(".alert").at(0).find("h4").text()).toEqual("Test");
+    expect(
+      wrapper.find(".alert").at(0).contains("Have you seen the Fnords?")
+    ).toBeTruthy();
+
+    expect(wrapper.find(".alert").at(1).hasClass("alert-danger")).toBeTruthy();
+    expect(
+      wrapper.find(".alert").at(1).contains("The foo hit the bar")
+    ).toBeTruthy();
+
+    expect(wrapper.find(".alert").at(2).hasClass("alert-success")).toBeTruthy();
+    expect(
+      wrapper.find(".alert").at(2).contains("Time for *us* to hit the bar")
+    ).toBeTruthy();
+  });
+
+  describe("newAlert", () => {
+    it("creates a new alert", () => {
+      const WrappedComponent = withAlertNotifications(fakeComponent);
+      const wrapper = shallow<React.ComponentType>(<WrappedComponent />);
+      const instance = wrapper.instance();
+
+      expect(wrapper.state().alerts).toEqual([]);
+
+      (instance as any).newAlert("warning", "Test", "foobar");
```

Comment:
I can't seem to make typescript recognize that this component implements `AlertNotificationsI` and thus the `newAlert` and `onAlertDismissed` methods.
Ah well.